lua/utils.lua

```diff
@@ -112,24 +122,32 @@ end
 
 ---@param dir string
 M.open_dir = function(dir)
-    if inside_tmux then
-        local open_cmd = string.format('tea %s', dir)
-        local open_result = os.execute(open_cmd)
-        if open_result == 0 then
-            return
-        end
+  if M.open_dir_provider ~= "telescope" and M.open_dir_provider ~= "fzf_lua" then
+    error("Invalid `open_dir_provider`: " .. M.open_dir_provider .. "\nPlease use either 'telescope' or 'fzf_lua'.")
+  end
+
+  if inside_tmux then
+    local open_cmd = string.format('tea %s', dir)
+    local open_result = os.execute(open_cmd)
+    if open_result == 0 then
+      return
     end
-    vim.schedule(function()
-        vim.cmd('cd ' .. dir)
-
-        local is_git_repo = vim.fn.system('git rev-parse --is-inside-work-tree 2>/dev/null'):match('true')
-
-        if is_git_repo then
-            vim.cmd('Telescope git_files cwd=' .. dir)
-        else
-            vim.cmd('Telescope find_files')
-        end
-    end)
+  end
+  vim.schedule(function()
+    vim.cmd('cd ' .. dir)
+
+    local is_git_repo = vim.fn.system('git rev-parse --is-inside-work-tree 2>/dev/null'):match('true')
+
+    if is_git_repo and M.open_dir_provider == "telescope" then
+      vim.cmd('Telescope git_files cwd=' .. dir)
+    elseif is_git_repo and M.open_dir_provider == "fzf_lua" then
+      vim.cmd('FzfLua git_files cwd=' .. dir)
+    elseif M.open_dir_provider == "telescope" then
+      vim.cmd('Telescope find_files')
+    else
+      vim.cmd('FzfLua files')
+    end
```

Comment:
might be useful to simplify this with a map and helper function, something like this

```lua
local function get_open_command(is_git_repo, dir)
  local commands = {
    git = {
      telescope = 'Telescope git_files cwd=',
      fzf_lua = 'FzfLua git_files cwd='
    },
    no_git = {
      telescope = 'Telescope find_files',
      fzf_lua = 'FzfLua files'
    }
  }

  if is_git_repo then
    return commands.git[M.fuzzy_provider] .. (dir or '')
  else
    return commands.no_git[fuzzy_provider]
  end
end

-- and then in open_dir, we could just do
vim.cmd(get_open_command(is_git_repo, dir))
```

Reply:
Makes sense - refactored to use such a map and helper function instead :)